dotCMS/src/main/java/com/dotcms/analytics/metrics/QueryParameterValuesGetter.java

```diff
@@ -0,0 +1,64 @@
+package com.dotcms.analytics.metrics;
+
+import com.dotcms.analytics.metrics.AbstractCondition.AbstractParameter;
+import com.dotcms.experiments.business.result.Event;
+import com.dotmarketing.exception.DotRuntimeException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * {@link ParameterValueGetter} to get the Query Parameters from the Event's Url.
+ */
+public class QueryParameterValuesGetter implements ParameterValueGetter<QueryParameter> {
+    @Override
+    public Collection<QueryParameter> getValuesFromEvent(
+            final AbstractParameter parameter,
+            final Event event) {
+
+        final String urlValue = event.get("url").map(Object::toString).orElseThrow();
+
+        try {
+            URI uri = new URI(urlValue);
+            String query = uri.getQuery();
+
+            return getQueryParams(query).entrySet().stream()
+                    .map(entry -> new QueryParameter(entry.getKey(), entry.getValue()))
+                    .collect(Collectors.toList());
+        } catch (URISyntaxException e) {
+            throw new DotRuntimeException(e);
+        }
+    }
+
+    private static Map<String, String> getQueryParams(String query) {
```

Comment:
I wonder is there's already a library that takes care of the parsing of a http query.
If not, shouldn't these methods exist in "Utils" class?

Reply:
I think the `om.dotcms.util.HttpRequestDataUtil` has the method Freddy needs.

Reply:
done https://github.com/dotCMS/core/pull/25534/commits/604cb4212cb98ec878314c443a29eddd27185e21#diff-0be3d9cc6974385beca6556e8697c31c3a235292a46c189d54017ae50c2fdf39R7

the method that exists is this one 

https://github.com/dotCMS/core/blob/604cb4212cb98ec878314c443a29eddd27185e21/dotCMS/src/main/java/com/dotcms/util/HttpRequestDataUtil.java#L153

but it is not exactly what I need